infrastructure/godisk/disk_service.go

```diff
@@ -0,0 +1,93 @@
+package godisk
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/diskfs/go-diskfs"
+	"github.com/diskfs/go-diskfs/disk"
+	"github.com/diskfs/go-diskfs/filesystem"
+	units "github.com/docker/go-units"
+	"github.com/spf13/afero"
+
+	"github.com/weaveworks-liquidmetal/flintlock/core/ports"
+)
+
+func New(fs afero.Fs) ports.DiskService {
+	return &diskService{
+		fs: fs,
+	}
+}
+
+type diskService struct {
+	fs afero.Fs
+}
+
+// Create will create a new disk.
+func (s *diskService) Create(ctx context.Context, input ports.DiskCreateInput) error {
+	if input.Path == "" {
+		return errPathRequired
+	}
+	if input.Size == "" {
+		return errSizeRequired
+	}
+
+	if _, err := s.fs.Stat(input.Path); err == nil {
+		if removeErr := s.fs.Remove(input.Path); removeErr != nil {
+			return fmt.Errorf("removing disk %s: %w", input.Path, removeErr)
+		}
```

Comment:
can you explain this bit here: why do we delete whatever it at the input path first? it is to clean whatever is there before we create over it?

Reply:
Yep it's basically that. If the disk image exists already then delete and we will re-create.

What do you think to that default behaviour? Should i make it configurable? Perhaps add an option to `DiskCreateInput` valled `OverwriteExisting`?

Reply:
maybe.

worst case is someone sets Path to something they should not and it is erased. having it error unless told explicitly to overwrite may be useful

Reply:
Agreed, changing now...

Reply:
@Callisto13 - all changed.